vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -1623,15 +1623,47 @@ protected <U extends GridArrayUpdater, B extends DataCommunicatorBuilder<T, U>>
             Class<T> beanType,
             SerializableBiFunction<UpdateQueueData, Integer, UpdateQueue> updateQueueBuilder,
             B dataCommunicatorBuilder) {
+        this(beanType, updateQueueBuilder, dataCommunicatorBuilder, true);
+    }
+
+    /**
+     * Creates a new grid with an initial set of columns for each of the bean's
+     * properties. The property-values of the bean will be converted to Strings.
+     * Full names of the properties will be used as the
+     * {@link Column#setKey(String) column keys} and the property captions will
+     * be used as the {@link Column#setHeader(String) column headers}.
+     * <p>
+     * When autoCreateColumns is <code>true</code>, only the direct properties
+     * of the bean are included and they will be in alphabetical order. Use
+     * {@link Grid#setColumns(String...)} to define which properties to include
+     * and in which order. You can also add a column for an individual property
+     * with {@link #addColumn(String)}. Both of these methods support also
+     * sub-properties with dot-notation, eg.
+     * <code>"property.nestedProperty"</code>.
+     *
+     * @param beanType
+     *            the bean type to use, not <code>null</code>
+     * @param updateQueueBuilder
+     *            the builder for new {@link UpdateQueue} instance
+     * @param dataCommunicatorBuilder
+     *            Builder for {@link DataCommunicator} implementation this Grid
+     *            uses to handle all data communication.
+     * @param <B>
+     *            the data communicator builder type
+     * @param <U>
+     *            the GridArrayUpdater type
+     * @param autoCreateColumns
+     *            when <code>true</code>, columns are created automatically for
+     *            the properties of the beanType
+     */
+    protected <U extends GridArrayUpdater, B extends DataCommunicatorBuilder<T, U>> Grid(
+            Class<T> beanType,
+            SerializableBiFunction<UpdateQueueData, Integer, UpdateQueue> updateQueueBuilder,
+            B dataCommunicatorBuilder, boolean autoCreateColumns) {
         this(50, updateQueueBuilder, dataCommunicatorBuilder);
-        Objects.requireNonNull(beanType, "Bean type can't be null");
         Objects.requireNonNull(dataCommunicatorBuilder,
                 "Data communicator builder can't be null");
-        this.beanType = beanType;
-        propertySet = BeanPropertySet.get(beanType);
-        propertySet.getProperties()
-                .filter(property -> !property.isSubProperty())
-                .forEach(this::addColumn);
+        configureBeanType(beanType, autoCreateColumns);
```

Comment:
Good catch, it makes sense. There seems to be 2 options:

- Leave sorting related behaviour unaltered. This can be done by simply adding a private overload since it is only used in one place and it is in `Grid`
- Extract the behaviour altering PR an align all bean type configuration usages

What do you think is the best? Aligning the behaviour vs not breaking how it works?

Reply:
I'm inclined to prefer the second option. The lack of sorting in this place seems like an oversight, so a separate fix would make sense to me.

Reply:
It seems that the sorting logic has already been [removed and moved to Flow](https://github.com/vaadin/flow-components/pull/6551). So I guess behavior altering PR extraction is not necessary anymore.

Reply:
Oh, great!